modules/openapi-generator/src/main/resources/python/api.mustache

```diff
@@ -35,307 +40,237 @@ class {{classname}}:
+{{>partial_api}}
+        response_data = {{#asyncio}}await {{/asyncio}}self.api_client.call_api(
+            *_param,
+            _request_timeout=_request_timeout
+        )
+        {{#asyncio}}await {{/asyncio}}response_data.read()
+        return self.api_client.response_deserialize(
+            response_data=response_data,
+            response_types_map=_response_types_map,
+        )
 
-{{#allParams}}
-        :param {{paramName}}:{{#description}} {{{.}}}{{/description}}{{#required}} (required){{/required}}{{#optional}}(optional){{/optional}}
-        :type {{paramName}}: {{dataType}}{{#optional}}, optional{{/optional}}
-{{/allParams}}
-{{^asyncio}}
-        :param async_req: Whether to execute the request asynchronously.
-        :type async_req: bool, optional
-{{/asyncio}}
-        :param _request_timeout: timeout setting for this request.
-               If one number provided, it will be total request
-               timeout. It can also be a pair (tuple) of
-               (connection, read) timeouts.
-        :return: Returns the result object.
-                 If the method is called asynchronously,
-                 returns the request thread.
-        :rtype: {{returnType}}{{^returnType}}None{{/returnType}}
-        """
-        kwargs['_return_http_data_only'] = True
-        if '_preload_content' in kwargs:
-            message = "Error! Please call the {{operationId}}_with_http_info method with `_preload_content` instead and obtain raw data from ApiResponse.raw_data"  # noqa: E501
-            raise ValueError(message)
-
-        return {{#asyncio}}await {{/asyncio}}self.{{operationId}}_with_http_info.raw_function(
-            {{#allParams}}
-            {{paramName}},
-            {{/allParams}}
-            **kwargs,
+
+    @validate_call
+    {{#asyncio}}async {{/asyncio}}def {{operationId}}_without_preload_content{{>partial_api_args}} -> RESTResponseType:
+{{>partial_api}}
+        response_data = {{#asyncio}}await {{/asyncio}}self.api_client.call_api(
+            *_param,
+            _request_timeout=_request_timeout
+        )
+        return response_data.response
+
+
+    {{^asyncio}}
+    @validate_call
+    def {{operationId}}_with_async{{>partial_api_args}} -> ApplyResult:
+{{>partial_api}}
+        def callback(*args):
+            response_data = self.api_client.call_api(*args)
+            response_data.read()
+            return self.api_client.response_deserialize(
+                response_data=response_data,
+                response_types_map=_response_types_map,
+            ).data
+        return self.api_client.pool.apply_async(
+            callback,
+            _param + (_request_timeout,)
         )
 
+
     @validate_call
-    {{#asyncio}}async {{/asyncio}}def {{operationId}}_with_http_info(
+    def {{operationId}}_with_http_info_async{{>partial_api_args}} -> ApplyResult:
```

Comment:
@multani and I both suggested removing the `multiprocessing` async functionality in https://github.com/OpenAPITools/openapi-generator/issues/16788#issuecomment-1760950037

Now is a very good time for this, because anyone who previously relied on the `async_req` flag will have to touch all their code anyway. Just highlighting this so a clear decision is made.

Reply:
Removed multiprocessing. 4d3d312aee20229aa3cbf7406ec722843d8d6f8a